### PR TITLE
Perf test users/{user id} issue#73

### PR DESCRIPTION
**Pull request by StopMakingUsernamesUnique**
Add a custom github action and associated test yaml  file using AWS access and secret access keys to create lambdas to ping users/{userID} 50 times per second for 200 seconds

### CODE REVIEW

File: .github/workflows/performancetest-users-userID.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      # once a folder with slsart script.yml and serverless.yml are created and in the repo
+      # we can copy them into the working dir to run those specific tests with those configurations
+      # - currently this just runs the defaults
+      # examples for future:
+      # - run: cp ./slsart/script.yml ./
+      # - run: cp ./slsart/serverless.yml ./
+      - name: Deploy, invoke, remove functions
+        run: |
+          ~/.npm-global/bin/slsart deploy --region $AWS_SLSART_REGION
+          ~/.npm-global/bin/slsart invoke -s ../../test/artillery/performanceScript-users-userID.yaml --region $AWS_SLSART_REGION  > ../../test/results/performanceResults-users-userID.json
```
Comment (zxkevinxz) [inline .github/workflows/performancetest-users-userID.yaml:41]: The paths need to be changed, they are causing an error. It is determined from where it is being typed in the console, not where the file is. The command would be ran at the root of the console, so you can just remove the ../.. and just have /test...

**Review by zxkevinxz — CHANGES_REQUESTED** [review]
Looks good but the paths I mention below need to be changed.

File: .github/workflows/performancetest-users-userID.yaml (hunk 1 above)
Comment (StopMakingUsernamesUnique) [reply .github/workflows/performancetest-users-userID.yaml:41]: Good to know, thanks for testing!

**Comment by chonger878** [comment]
I feel like I can follow through the code easily and understand how it works.  The only thing that slightly (like very slightly) threw me off was line 38 - 40ish of the performancetest-users-userID.yaml file, when I saw there was nothing after the dash, but then I noticed the expression/function/whatever the step is on the following line.  Is there a way I can test the code if it's not merged with the practicum's repository?

**Comment by chonger878** [comment]
Never mind, my yaml file has the same thing too.  Disregard my remark about that!

File: .github/workflows/performancetest-users-userID.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      # once a folder with slsart script.yml and serverless.yml are created and in the repo
+      # we can copy them into the working dir to run those specific tests with those configurations
+      # - currently this just runs the defaults
+      # examples for future:
+      # - run: cp ./slsart/script.yml ./
+      # - run: cp ./slsart/serverless.yml ./
+      - name: Deploy, invoke, remove functions
+        run: |
+          ~/.npm-global/bin/slsart deploy --region $AWS_SLSART_REGION
+          ~/.npm-global/bin/slsart invoke -s test/artillery/performanceScript-users-userID.yaml --region $AWS_SLSART_REGION  > test/results/performanceResults-users-userID.json
```
Comment (zxkevinxz) [inline .github/workflows/performancetest-users-userID.yaml:41]: Looks like you need to change the -s flag for a script
Screen Shot 2021-02-04 at 10 42 30 PM

**Review by zxkevinxz — CHANGES_REQUESTED** [review]

File: test/artillery/performanceScript-users-userID.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+# You can find great documentation of the possibilities at:
+# https://artillery.io/docs/
+config:
+  # this hostname will be used as a prefix for each URI in the flow unless a complete URI is specified
+  target: "http://nsc.functionsapp-team1.azurewebsites.net/api/"
+  phases:
+    
+      -duration: 50 #warmup
+       arrivalRate: 5
+        variables:
```
Comment (zxkevinxz) [inline test/artillery/performanceScript-users-userID.yaml:13]: variables needs to be lined up with arrivalRate

File: test/artillery/performanceScript-users-userID.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+# This default script is intended to get you started quickly.
+# There is a lot more that Artillery can do.
+# You can find great documentation of the possibilities at:
+# https://artillery.io/docs/
+config:
+  # this hostname will be used as a prefix for each URI in the flow unless a complete URI is specified
+  target: "http://nsc.functionsapp-team1.azurewebsites.net/api/"
+  phases:
+    
+      -duration: 50 #warmup
```
Comment (zxkevinxz) [inline test/artillery/performanceScript-users-userID.yaml:11]: I think you need a space between the hyphen and duration

File: test/artillery/performanceScript-users-userID.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+              - "2"
+              - "3"
+              - "5"
+              - "6"
+         - userName:
+              - "Jorden"
+              - "Aidan"
+              - "Macaulay"
+              - "Leroy"
+              - "Naomi"
```
Comment (zxkevinxz) [inline test/artillery/performanceScript-users-userID.yaml:36]: Are the userNames needed?

File: test/artillery/performanceScript-users-userID.yaml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  -
+    flow:
+
+      - get:
+          url: "/users/{{user}}"
+
+          #      - update:
+          # url: "/users/{{user}}"
+          # json:
+          #    firstname: "{{userName}}"
```
Comment (zxkevinxz) [inline test/artillery/performanceScript-users-userID.yaml:47]: can this be deleted?

**Review by zxkevinxz — CHANGES_REQUESTED** [review]

**Review by chonger878 — APPROVED** [review]
Like I mentioned, I definitely felt like this was easy to follow and apologies for my misleading comment about that dash.

File: test/artillery/performanceScript-users-userID.yaml (hunk 6 above)
Comment (StopMakingUsernamesUnique) [reply test/artillery/performanceScript-users-userID.yaml:47]: I was planning on working on this piece after we confirm that the normal test works

File: test/artillery/performanceScript-users-userID.yaml (hunk 5 above)
Comment (StopMakingUsernamesUnique) [reply test/artillery/performanceScript-users-userID.yaml:36]: Usernames are related to update test, might drop however for time constraints.

**Review by zxkevinxz — APPROVED** [review]
This looks better, good work on the fixes. I know the endpoint is down so you will not be able to test it until those are update.

**Pull request merged**